pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -282,12 +332,14 @@ func (r *rangeAllocator) AllocateOrOccupyCIDR(logger klog.Logger, node *v1.Node)
 
 	//queue the assignment
 	logger.V(4).Info("Putting node with CIDR into the work queue", "node", klog.KObj(node), "CIDRs", allocated.allocatedCIDRs)
-	r.nodeCIDRUpdateChannel <- allocated
-	return nil
+	return r.updateCIDRsAllocation(logger, allocated)
```

Comment:
we can improve the `updateCIDRsAllocation` function to receive the node object and the cidrstrings and delete the  `nodeReservedCIDRs ` struct

Reply:
Updated.